### ISSUE TITLE
Peerswap plugin killed on node startup 

### ISSUE THREAD

**Issue by ShahanaFarooqui** [ElementsProject/peerswap#140]
Peerswap plugin on Core Lightning is being killed on startup for the node with below error:

```
2022-10-05T17:16:56.975Z DEBUG   plugin-peerswap-plugin: poll_service: could not send request_poll msg: -32600:Peer is not connected
2022-10-05T17:16:56.975Z DEBUG   0391cd7ae2eca090c5f807d71ab600bea5590796b56b21b44de596f9be42b7fd96-connectd: peer_out INVALID 42085
2022-10-05T17:16:56.979Z INFO    plugin-peerswap-plugin: [Swap:96500ce1c7824710450a325ee0e9894d619ecfe0d666abe8300e8ef53183aea1]: Recovering from state State_SwapOutSender_AwaitAgreement
2022-10-05T17:16:56.980Z DEBUG   plugin-peerswap-plugin: [FSM] event:id: 96500ce1c7824710450a325ee0e9894d619ecfe0d666abe8300e8ef53183aea1, Event_ActionFailed on State_SwapOutSender_AwaitAgreement
2022-10-05T17:16:56.980Z INFO    plugin-peerswap-plugin: plugin quitting, error: event rejected
2022-10-05T17:16:56.981Z INFO    plugin-peerswap-plugin: Killing plugin: exited during normal operation
```

CLN node config:
```
network=signet
log-level=debug
bitcoin-rpcuser=***
bitcoin-rpcpassword=***
plugin=/.../peerswap/peerswap-plugin
peerswap-db-path=/.../l1-signet/swap
peerswap-policy-path=/.../l1-signet/policy.conf
``` 

I am unable to start peerswap on this node anymore.

**Comment by ShahanaFarooqui** [comment]
@nepet 
1: This issue is persistent even without changing networks (as oppose to our discussion in today's meeting). 
2: Steps to reproduce:
- Start CLN with Peerswap plugin
- Execute a successful `swap-out`/`swap-in` from the node.
- Stop CLN while this swap is in active state.
- Restart CLN. 
- peerswapd will crash with error message `Killing plugin: exited during normal operation`.

**Comment by nepet** [comment]
More insight: This seems to be the case when recovering from a  State_SwapInSender_AwaitClaimPayment. On recovery the following nil pointer exception is recorder
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xbe91f4]

goroutine 1 [running]:
github.com/elementsproject/peerswap/txwatcher.(*BlockchainRpcTxWatcher).AddWaitForCsvTx(0xc0005266c0, {0xc00011f2c0, 0x40}, {0xc00011ec80, 0x40}, 0x0, 0xaf, {0xc000540200, 0x22, 0x1f4})
	/home/pn/Projects/Git/peerswap/txwatcher/rpctxwatcher.go:265 +0xf4
github.com/elementsproject/peerswap/swap.(*AwaitPaymentOrCsvAction).Execute(0xce6720?, 0xc0001302a0, 0xc0001cf080)
	/home/pn/Projects/Git/peerswap/swap/actions.go:312 +0x26b
github.com/elementsproject/peerswap/swap.(*SwapStateMachine).Recover(0xc0002d81c0)
	/home/pn/Projects/Git/peerswap/swap/fsm.go:233 +0x1ab
github.com/elementsproject/peerswap/swap.(*SwapService).RecoverSwaps(0xc00011df20)
	/home/pn/Projects/Git/peerswap/swap/service.go:131 +0x4e7
main.run()
	/home/pn/Projects/Git/peerswap/cmd/peerswap-plugin/main.go:311 +0x12a5
main.main()
	/home/pn/Projects/Git/peerswap/cmd/peerswap-plugin/main.go:57 +0xe5
```

**Comment by nepet** [comment]
This might be related to #73

**Comment by ShahanaFarooqui** [comment]
Let me know if you would prefer me to close this issue after the fix is merged with PR #155.
